python-avd/pyavd/j2filters/natural_sort.py

```diff
@@ -58,11 +58,11 @@ def alphanum_key(key: Any) -> list:
                 msg = f"Missing key '{sort_key}' in item to sort {key}."
                 raise KeyError(msg)
             return [convert(c, ignore_case) for c in re.split(pattern, str(key.get(sort_key, key)))]
-        if sort_key is not None and isinstance(key, Namespace):
+        if sort_key is not None and (isinstance(key, Namespace) or hasattr(key, "_is_avd_model")):
```

Comment:
cannot use `isinstance(key, (Namespace, AvdModel))`? (maybe circular imports?

Reply:
Since this is a j2 filter I would prefer not to import our schema models here. With the latest list refactor I hope I can get rid of this again, since the AvdList has the ._natural_sorted() method.

Reply:
I removed the check since we no longer use this. Also updated the function type hints to be more neutral (Iterable, Mapping etc)